Improve addition assignment example

SMYusupov opened the pull request:
#### Summary
Fix expected result, make code more readable

#### Motivation
Make code more readable and don't be misled

This PR…
-->
- [ ] Make expected result correct
- [ ] Rewrite code to make it more readable

SMYusupov committed:
Update index.md

wbamberg reviewed (changes requested):
Thanks for your PR, @P1GAS !

I agree that we could clarify this example and make it more "real".

But this change is not quite right. I think the existing page is expecting that each assignment is being made on the original value of the variable.

For example, here:

```js
// Boolean + Number -> addition
baz += 1 // 2

// Boolean + Boolean -> addition
baz += false // 2
```

...in the second addition assignment, it says `baz` is boolean, but after the first one it is a number.

Similarly, your change here:

```js
baz += false // 2
```

...seems to be because you previously incremented `baz`, but that's inconsistent with the later one:

```js
let foo = 'foo'

// String + Boolean -> concatenation
foo += false // "foofalse"

// String + String -> concatenation
foo += 'bar' // "foobar"
```

...why isn't the last value here "foofalsebar"?

I think it would be best to make it very clear that these are all self-contained examples, perhaps by having separate code blocks for them? For example:

```js
let bar = 5;

// Number + Number -> addition
bar += 2; // 7
```

```js
let bar = 5;

// Number + String -> concatenation
bar += 'foo'; // "5foo"
```
...etc

SMYusupov committed:
Update index.md

P1GAS commented:
Hello @wbamberg! I have changed the code, could you check it again?

wbamberg reviewed (changes requested):
Thank you for the update @P1GAS . But this isn't the change I asked for. I suggested making each assignment a separate self-contained example. Is there a reason you don't think this is a good idea?

P1GAS commented:
I think that's a good idea. I have another view similar to yours. What if we create 3 blocks with every variable

wbamberg commented:
> @wbamberg I didn't understand you previously. I think that's a good idea. I have another view similar to yours. What if we create 3 blocks for every variable
> 

Yes, this looks good to me too. But please use `const` rather than `let` here, and semicolons at the ends of lines.

teoli2003 commented:
> > @wbamberg I didn't understand you previously. I think that's a good idea. I have another view similar to yours. What if we create 3 blocks for every variable
> 
> Yes, this looks good to me too. But please use `const` rather than `let` here, and semicolons at the ends of lines.

These must be let as the operator `+=` is used. But I agree for the semi-colon.

wbamberg commented:
> > > @wbamberg I didn't understand you previously. I think that's a good idea. I have another view similar to yours. What if we create 3 blocks for every variable
> > 
> > 
> > Yes, this looks good to me too. But please use `const` rather than `let` here, and semicolons at the ends of lines.
> 
> These must be let as the operator `+=` is used. But I agree for the semi-colon.

Doh, sorry! Yes of course.

SMYusupov committed:
Update index.md

make the example more readable & replace bar to bottom, it's more difficult because it makes addition and concatenation

SMYusupov committed:
Update index.md

add semicolon

Josh-Cena commented on files/en-us/web/javascript/reference/operators/addition_assignment/index.md:
````diff
@@ -21,36 +21,41 @@ possible.
 
 ```js
-x += y // x = x + y
+x += y; // x = x + y
````
```suggestion
x += y // x = x + y
```

Josh-Cena committed:
Update files/en-us/web/javascript/reference/operators/addition_assignment/index.md

wbamberg reviewed (approved):
👍 looks good to me, thank you @P1GAS !
